tests/unit/test_retry_policy.py

```diff
@@ -0,0 +1,62 @@
+from unittest import mock
+
+import pytest
+
+from globus_sdk.transport import RetryContext, RetryPolicy
+
+
+def test_retry_policy_respects_retry_after():
+    policy = RetryPolicy()
+
+    dummy_response = mock.Mock()
+    dummy_response.headers = {"Retry-After": "5"}
+    dummy_response.status_code = 429
+    ctx = RetryContext(1, retry_state={}, response=dummy_response)
+
+    with mock.patch("time.sleep") as mocksleep:
```

Comment:
The `mocksleep` fixture could be in a higher-level conftest to also use here

Reply:
Good call, I'll make that change.